test/query/shorthand.test.ts

```diff
@@ -33,6 +35,144 @@ describe('query/shorthand', () => {
     });
   });
 
+  describe('parse', () => {
+    it('should correctly parse a shorthand string', () => {
+      let specQ = parse(
+        'point|calculate:{b2:3*datum["b2"]}|filter:"datum[\\"b2\\"] > 60"|filterInvalid:false|x:b2,q|y:bin(balance,q,sort={"field":"balance","op":"mean","order":"descending"})'
+      );
+
+      assert.deepEqual(specQ, {
+        transform: {
+          calculate: [{field: 'b2', expr: '3*datum["b2"]'}],
+          filter: 'datum["b2"] > 60',
+          filterInvalid: false
+        },
+        mark: Mark.POINT,
+        encodings: [
+          {channel: Channel.X, field: 'b2', type: Type.QUANTITATIVE},
+          {bin: {}, channel: Channel.Y, field: 'balance', type: Type.QUANTITATIVE, sort: {field: 'balance', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}}
+        ]
+      });
+    });
+
+    it('should correctly parse an ambiguous shorthand', () => {
+      let specQ = parse('?|?:?(?,?)');
```

Comment:
Actually never mind ... my updated function in #262 would require rewriting this part anyway.